src/transport_ofi.h

```diff
@@ -308,6 +310,17 @@ extern struct fid_ep* shmem_transport_ofi_target_ep;
             shmem_free_list_unlock(ctx->bounce_buffers);                        \
     } while (0)
 
+static inline
+void shmem_transport_probe(void) {
+#if defined(ENABLE_MANUAL_PROGRESS) && !defined(HARD_POLLING)
+    if (0 == pthread_mutex_trylock(&shmem_transport_ofi_progress_lock)) {
```

Comment:
The lock is only required when `USE_THREAD_COMPLETION` is defined.  Not sure if there is any benefit from having a single thread poke OFI when `FI_THREAD_SAFE` is used.